Realm/RLMQueryUtil.mm

```diff
@@ -1850,6 +1848,45 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                        @"Only support compound, comparison, and constant predicates");
     }
 }
+
+// This function returns the nested subscripts from a NSPredicate with the following format `anyCol[0]['key']['all']`
+// and its respective keypath (including any linked keypath)
+// This will iterate each argument of the NSExpression and its nested NSExpressions, takes the constant subscript
+// and creates a PathElement to be used in the query.
+NSString* QueryBuilder::get_path_elements(std::vector<PathElement> &paths, NSExpression *expression) {
+    NSString *keyPath = @"";
+    for (NSUInteger i = 0; i < expression.arguments.count; i++) {
+        NSString *nestedKeyPath = @"";
+        if (expression.arguments[i].expressionType == NSFunctionExpressionType) {
+            nestedKeyPath = get_path_elements(paths, expression.arguments[i]);
+        } else if (expression.arguments[i].expressionType == NSConstantValueExpressionType) {
+            id value = [expression.arguments[i] constantValue];
+            if ([value isKindOfClass:[NSNumber class]]) {
+                paths.push_back(PathElement{[(NSNumber *)value intValue]});
+            } else if ([value isKindOfClass:[NSString class]]) {
+                NSString *key = (NSString *)value;
+                if ([key isEqual:@"*"]) {
+                    paths.push_back(PathElement{});
+                } else {
+                    paths.push_back(PathElement{key.UTF8String});
+                }
+            } else {
+                throwException(@"Invalid subscript type",
+                               @"Only `Strings` or index are allowed subscripts");
+            }
+        } else if (expression.arguments[i].expressionType == NSKeyPathExpressionType) {
+            nestedKeyPath = expression.arguments[i].keyPath;
+        } else {
+            throwException(@"Invalid expression type",
+                           @"Subscripts queries don't allow any other expression types");
+        }
+        if ([nestedKeyPath length] > 0) {
+            keyPath = ([keyPath length] > 0) ? [NSString stringWithFormat:@"%@.%@", keyPath, nestedKeyPath] : [NSString stringWithFormat:@"%@", nestedKeyPath];
```

Comment:
`nestedKeyPath` already is a string and `[NSString stringWithFormat:@"%@", nestedKeyPath]` is just a complicated no-op.